File: 044-扑克牌顺子/044-IsContinuous.py
# -*- coding:utf-8 -*-
class Solution:
    def IsContinuous(self, numbers):
        # write code here
        if len(numbers)<5:
            return False
        pool=set()
        mi=float('inf')
        ma=float('-inf')
        for val in numbers:
            if val!=0:
                if val in pool:
                    return False
                else:
                    pool.add(val)
                    mi=min(mi,val)
                    ma=max(ma,val)
        if ma-mi>4:
            return False
        return True

File: 044-扑克牌顺子/044-IsContinuous.cpp
class Solution {
public:
    bool IsContinuous( vector<int> numbers ) {
        if(numbers.size()<5){
            return false;
        }
        int max=INT_MIN;
        int min=INT_MAX;
        vector<int> map(14,0);
        for(auto val:numbers){
            if (val!=0){
                if (map[val]==1){
                    return false;
                }
                else{
                    map[val]=1;
                }
                max=max>val?max:val;
                min=min<val?min:val;
            }
        }
        if((max-min) >4){
            return false;
        }
        return true;
    }
};
